app/model/scoreboard.js

```diff
@@ -0,0 +1,135 @@
+import Target from './target.js';
+import Round from './round.js';
+
+
+/**
+ * Keeps track of the scoreboard and it's affiliated stats
+ */
+export default class Scoreboard {
+
+    constructor() {
+        let targetTypes = ['20', '18', '19', '17', '16', '15', 'B'];
+        this._targets = [];
+        for(let targetType of targetTypes) {
+            this._targets.push({type: targetType, counter: new Target()});
+        }
+        this._activeTargetIndex = 0;
+        this._activeTarget = this._targets[this._activeTargetIndex];
+
+        this._round = new Round();
+    }
+
+    /**
+     * Marks a dart that hits. Moves to next target as appropriate.
+     * Ignored if scoreboard is already done.
+     * @returns {Scoreboard}
+     */
+    hit() {
+        if(!this.isDone()) {
+            this._round.nextDart();
+            this._activeTarget.counter.markHit({round: this._round.toString()});
```

Comment:
agreed. do you have any idea how hard it is to copy an object in javascript?

me either. i just know it's hard. I'll let you know how easy it is soon...